voltixApp/VoltixApp/Services/CryptoPriceService.swift

```diff
@@ -19,43 +16,49 @@ public class CryptoPriceService: ObservableObject {
 		let elapsedTime = Date().timeIntervalSince(cacheEntry.timestamp)
 		return elapsedTime <= 3600 // 1 hour in seconds
 	}
+    
+    func getPrice(priceProviderId: String) async -> Double {
+        print("Current currency: \(SettingsViewModel.shared.selectedCurrency.description())")
```

Comment:
`SettingsViewModel.shared.selectedCurrency.description()` , this is not correct , the selected currency should be saved into  `UserDefault`  when user change it in `SettingsViewModel` 

Here we should just read from UserDefault

Reply:
UserDefaults implemented.